$OpenBSD: patch-src_assignments_c,v 1.1 2015/12/18 15:53:09 dcoppa Exp $

Fixed logging statement.

Mark assignment as run before executing it.

--- src/assignments.c.orig	Wed Sep 30 08:55:10 2015
+++ src/assignments.c	Fri Dec 18 13:58:58 2015
@@ -40,6 +40,13 @@ void run_assignments(i3Window *window) {
         if (skip)
             continue;
 
+        /* Store that we ran this assignment to not execute it again. We have
+         * to do this before running the actual command to prevent infinite
+         * loops. */
+        window->nr_assignments++;
+        window->ran_assignments = srealloc(window->ran_assignments, sizeof(Assignment *) * window->nr_assignments);
+        window->ran_assignments[window->nr_assignments - 1] = current;
+
         DLOG("matching assignment, would do:\n");
         if (current->type == A_COMMAND) {
             DLOG("execute command %s\n", current->dest.command);
@@ -53,11 +60,6 @@ void run_assignments(i3Window *window) {
 
             command_result_free(result);
         }
-
-        /* Store that we ran this assignment to not execute it again */
-        window->nr_assignments++;
-        window->ran_assignments = srealloc(window->ran_assignments, sizeof(Assignment *) * window->nr_assignments);
-        window->ran_assignments[window->nr_assignments - 1] = current;
     }
 
     /* If any of the commands required re-rendering, we will do that now. */
@@ -76,7 +78,7 @@ Assignment *assignment_for(i3Window *window, int type)
         if ((type != A_ANY && (assignment->type & type) == 0) ||
             !match_matches_window(&(assignment->match), window))
             continue;
-        DLOG("got a matching assignment (to %s)\n", assignment->dest.workspace);
+        DLOG("got a matching assignment\n");
         return assignment;
     }
 
